router-tests/header_propagation_test.go

```diff
@@ -97,20 +99,20 @@ func TestHeaderPropagation(t *testing.T) {
 		}
 	}
 
-	setSubgraphPropagateHeader := func(header, valA, valB string) testenv.SubgraphsConfig {
+	setSubgraphPropagateHeader := func(header string, valA, valB []string) testenv.SubgraphsConfig {
 		return testenv.SubgraphsConfig{
 			Employees: testenv.SubgraphConfig{
 				Middleware: func(handler http.Handler) http.Handler {
 					return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-						w.Header().Set(header, valA)
+						w.Header()[header] = valA
```

Comment:
Hi @cmtm 

Manually modifying the underlying map is not fully equal to [header.Set](https://pkg.go.dev/net/http#Header.Set), because the wrapper always uses canonical form of the header name

The simplest fix will be
```
w.Header()[textproto.CanonicalMIMEHeaderKey(header)] = valA
```

Reply:
We should write a test that verify it.

Reply:
Done. I wrote a test for it, but I'm not sure how the http.Header type handles canonicalization internally. If it canonicalizes when deserializing an http request, we might not be able to properly test the canonicalization with it. We'd need to use a lower level construct.

Reply:
Note that I didn't change the behavior of this handler, since it's okay if it doesn't do canonicalization.

Reply:
why it is okay? we should use canonical header names whenever it is possible

Reply:
If I understand this correctly, this is the test code which mocks a subgraph and returns the headers requested by the test. The goal of this test is to verify the behavior of the cosmo router. Ensuring it properly handles non-canonicalized headers send by the subgraph are part of the test.